includes/class-wcs-payment-tokens.php

```diff
@@ -102,29 +102,26 @@ public static function get_subscriptions_from_token( $payment_token ) {
 		$users_subscription_ids = WCS_Customer_Store::instance()->get_users_subscription_ids( $payment_token->get_user_id() );
 
 		if ( ! empty( $users_subscription_ids ) ) {
-			$meta_query = array(
-				array(
-					'key'   => '_payment_method',
-					'value' => $payment_token->get_gateway_id(),
-				),
-				array(
-					'key'   => '_requires_manual_renewal',
-					'value' => 'false',
-				),
-				array(
-					'value' => $payment_token->get_token(),
-				),
+			$subscription_ids = wcs_get_orders_with_meta_query(
+				[
+					'type'           => 'shop_subscription',
+					'status'         => [ 'wc-pending', 'wc-active', 'wc-on-hold' ],
+					'payment_method' => $payment_token->get_gateway_id(),
+					'meta_query'     => [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+						[
+							'key'   => '_requires_manual_renewal',
+							'value' => 'false',
+						],
+						[
+							'value' => $payment_token->get_token(),
+						],
+					],
+					'limit'          => -1,
+					'return'         => 'ids',
+					'post__in'       => $users_subscription_ids,
```

Comment:
Can you explain why we use 'post' here? I tried to use `order__in` and it doesn't work.

Reply:
The `wc_get_orders()` function which we use in `wcs_get_orders_with_meta_query()` uses the `WP_Query` class for CPT and `OrdersTableQuery` when HPOS is enabled to query for orders by the given arguments. Both of these classes support the `post__in` argument.